### PR TITLE
Introduce CT for PHP7

### PR DESCRIPTION
**Pull request by keradus**
ref #1119, #2090

Idea defined by utests.
~~Implementation is missing.~~

### CODE REVIEW

**Comment by keradus** [comment]
https://wiki.php.net/rfc/multiple-catch needs to be added

**Comment by keradus** [comment]
done, please review

**Comment by GrahamCampbell** [comment]
What's the status on this please?

File: src/Tokenizer/Transformers.php [hunk 1]
```diff
@@ -90,11 +95,25 @@ public function hasCustomToken($value)
...
+        }
+    }
+
+    /**
      * Register Transformer.
      *
      * @param TransformerInterface $transformer Transformer
      */
-    public function registerTransformer(TransformerInterface $transformer)
+    private function registerTransformer(TransformerInterface $transformer)
```
Comment (GrahamCampbell) [inline src/Tokenizer/Transformers.php]: Maybe protected would be more suitable?

Comment (GrahamCampbell) [inline src/Tokenizer/Transformers.php]: Oh, nvm. I see this class is final. :)

File: src/Tokenizer/TransformerInterface.php [hunk 2]
```diff
@@ -34,7 +34,7 @@
     public function getCustomTokenNames();
 
     /**
-     * Return the name of the fixer.
+     * Return the name of the transformer.
```
Comment (GrahamCampbell) [inline src/Tokenizer/TransformerInterface.php]: :D

**Comment by GrahamCampbell** [comment]
This PR looks ok to me. Probably merge this and then see what reports come in. :)

**Comment by keradus** [comment]
Already marked as RTM.
Would be great to finish some other PRs and merge this after that.

**Comment by GrahamCampbell** [comment]
Or just merge this first since it's ready. :trollface:

**Comment by keradus** [comment]
easier to rebase this than fix upcoming code during 1.x->2.x merge.

**Comment by keradus** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;MedrL8rj&#x2F;27-new-ct-tokens"> new CT tokens</a>

**Comment by GrahamCampbell** [comment]
Board seems to be private. :P

image

**Comment by keradus** [comment]
no more comments like those should be made

**Comment by GrahamCampbell** [comment]
lol - i assumed you put that board there for us to see :)

**Comment by keradus** [comment]
updated plugin of service integration did that from mine account

**Comment by GrahamCampbell** [comment]
Ah. :D

**Pull request closed without merging**